common/v2/services/Store/StoreProvider.tsx

```diff
@@ -125,12 +125,18 @@ export const StoreProvider: React.FC = ({ children }) => {
   const [memberships, setMemberships] = useState<MembershipStatus[] | undefined>([]);
 
   const membershipState = (() => {
-    if (memberships) {
-      return Object.values(memberships).length > 0
-        ? MembershipState.MEMBER
-        : MembershipState.NOTMEMBER;
+    if (!memberships) {
+      return MembershipState.ERROR;
+    } else if (Object.values(memberships).length === 0) {
+      return MembershipState.NOTMEMBER;
+    } else {
+      const currentTime = Math.round(Date.now() / 1000);
+      if (membershipExpiration.some(expirationTime => expirationTime > currentTime)) {
+        return MembershipState.MEMBER;
+      } else {
+        return MembershipState.EXPIREDMEMBER;
```

Comment:
@blurpesec Will this ever be possible? I seem to remember you had some remarks regarding that we didn't detect expired memberships?

Reply:
From what I saw in the smart contract, the `keyExpirationTimestampFor(address)` function does return the correct uint256 expiration timestamp. This could be faked by changing the device time, but that's a calculated risk

Reply:
@MrLuit But I thought that didn't work when the actual lock was expired? Since we don't own the lock anymore then.